### PR TITLE
Datasource: Add support for configuring timeout for HTTP data sources

### LINKED ISSUE #3168: FR : Ability to configure timeout for querying datasources.
We should have a global config somewhere , where we can configure timeout settings for querying backend datasources in milliseconds/seconds.

### PR DESCRIPTION
**Pull request by smalukav**
Fixes #3168

Add timeout to datasource JsonData, use it in DataSource.GetHttpClient and DataSource.GetHttpTransport

### CODE REVIEW

**Comment by torkelo** [comment]
Think this looks pretty reasonable

**Review by achatterjee-grafana — CHANGES_REQUESTED** [review]

File: packages/grafana-ui/src/components/DataSourceSettings/DataSourceHttpSettings.tsx [hunk 1]
```diff
@@ -148,19 +148,34 @@ export const DataSourceHttpSettings: React.FC<HttpSettingsProps> = (props) => {
...
-                tags={dataSourceConfig.jsonData.keepCookies}
-                onChange={(cookies) =>
-                  onSettingsChange({ jsonData: { ...dataSourceConfig.jsonData, keepCookies: cookies } })
-                }
-              />
+            <div className="gf-form-group">
+              <div className="gf-form">
+                <InlineFormLabel
+                  width={11}
+                  tooltip="Grafana Proxy deletes forwarded cookies by default. Specify cookies by name that should be forwarded to the data source."
```
[the remark this answers is not in the document]
Comment (smalukav) [reply packages/grafana-ui/src/components/DataSourceSettings/DataSourceHttpSettings.tsx]: If was like that before but ok :)

**Review by achatterjee-grafana — APPROVED** [review]
LGTM :)

**Review by torkelo — COMMENTED** [review]
@bergquist should this need anything special considerations from Grafana Cloud? Thinking that uses can override the proxy timeout could cause issues?

**Comment by giovannirco** [comment]
I would love to see this merged, sometimes the database grows a lot and takes longer to finish, although I understand it starts to play the role of reporting, Grafana actually does an excellent job of showing some useful data from SQL, the only issue is that some queries takes longer and I can't control the timeout.

**Comment by smalukav** [comment]
@torkelo can it be merged in 7.6 by any chance?

**Comment by achatterjee-grafana** [comment]
I am not sure there will be a v7.6 version.

**Comment by smalukav** [comment]
> I am not sure there will be a v7.6 version.

8.0 would also work

**Review by aknuds1 — APPROVED** [review]

**Pull request merged**